### PR TITLE
DSD-1623: label spacing for DatePicker, Select, Slider, and TextInput

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1623](https://jira.nypl.org/browse/DSD-1623)

## This PR does the following:

- Updates the `DatePicker`, `Select`, `Slider`, and `TextInput` components to reduce the spacing between the field label and the field itself from `"8px"` to `"2px"`.
- NOTE: The `FilterBar` and `MultiSelectGroup` components were not addressed because they have not yet been reintroduced back into the DS.

## How has this been tested?

- local Storybook

## Accessibility concerns or updates

- n/a

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -225,6 +226,12 @@ const DatePickerWrapper: React.FC<
           isRequired={isRequired}
           legendText={labelText}
           showRequiredLabel={showRequiredLabel}
+          __css={{
+            legend: {
+              // marginBottom: "xxxs",
+              ...labelLegendTextSpecialSpacing,
+            },
```
Comment (EdwinGuzman) [inline src/components/DatePicker/DatePicker.tsx]: This can be `legend: labelLegendTextSpecialSpacing,`

File: src/theme/components/global.ts [hunk 2]
```diff
@@ -125,6 +125,10 @@ const labelLegendText = {
     color: "dark.ui.typography.heading",
   },
 };
+// Used in form inputs that require nuanced spacing.
+const labelLegendTextSpecialSpacing = {
+  marginBottom: "xxxs",
```
Comment (EdwinGuzman) [inline src/theme/components/global.ts:130]: Why not add this rule to `fieldset.ts` and `label.ts`? This seems like it'll be easier to miss if the update is in the component file than the style file.

Comment (bigfishdesign13) [reply src/theme/components/global.ts:130]: I was following the pattern of `labelLegendText` because I wanted a reusable object.

Comment (EdwinGuzman) [reply src/theme/components/global.ts:130]: Yes, but `labelLegendText` is used in `fieldset.ts` and `label.ts` and not in the `__css` prop for the components. So what's here is fine but if we can keep the update in the theme files, that'd be ideal.

**Review by EdwinGuzman — APPROVED** [review]
Approving but would like to see this in the theme file at some point if possible.

File: src/theme/components/datePicker.ts [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
  */
 
 const { defineMultiStyleConfig, definePartsStyle } =
-  createMultiStyleConfigHelpers(["subLabels"]);
+  createMultiStyleConfigHelpers(["fieldset", "subLabels"]);
 
 const DatePicker = defineMultiStyleConfig({
   baseStyle: definePartsStyle({
+    fieldset: {
+      legend: { ...labelLegendTextSpecialSpacing },
```
Comment (EdwinGuzman) [inline src/theme/components/datePicker.ts]: This can be `legend: labelLegendTextSpecialSpacing,`

**Review by EdwinGuzman — COMMENTED** [review]
Update looks good but just needs one minor update.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**